### PR TITLE
Update DG with details of export implementation

### LINKED ISSUE #102: Update DG with details of export implementation
Add description of the implementation of export feature in DG

### PR DESCRIPTION
**Pull request by yanchenglee98**
Add description of the implementation of export feature in DG
Closes #102

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -194,14 +194,42 @@ This section describes some noteworthy details on how certain features are imple
...
-* .txt files can be easily opened on most operating systems e.g. MacOS, Windows.
+##### Aspect: What file format to export to
+* **Alternative 1 (current choice):** Write to a .txt file.
+  * Pros: More user-friendly.
+  * Pros: Most operating system is able to open .txt files natively.
+  * Cons: Does not offer much functionality apart from viewing and copying.
+
+* **Alternative 2:** Write to a .json file
+  * Pros: More well known among developers.
+  * Cons: Less technical users may not know how to open a .json file.
```
Comment (schoolex) [inline docs/DeveloperGuide.md]: Better to use a table form

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -194,14 +194,42 @@ This section describes some noteworthy details on how certain features are imple
...
+
+Given below is a step-by-step usage scenario of how the `export` feature works:
+
+Step 1. The user launches the application and inputs `export email` into the input box.
+
+Step 2. The `LogicManager#execute()` is then called and the input is parsed through `AddressBookParser#parseCommand()`, returning an `ExportCommand`.
+
+Step 3. The `export` command then calls `ExportCommand#execute()`, and calls `Model#getAddressBook()` follwed by `ReadOnlyAddressBook#getPersonList()` to get the current list of persons.
+
+Step 4. The person list is then passed to `ExportCommand#handlEmail()` which iterates through the list and calls `Person#getEmail()` to access the `Email` and writes to the file `hally.txt`
```
Comment (schoolex) [inline docs/DeveloperGuide.md]: Is the format consistent with the numbering style in the logic API?

**Review by schoolex — COMMENTED** [review]
There are still some design format that we still need to discuss

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (yanchenglee98) [reply docs/DeveloperGuide.md]: OK done

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (yanchenglee98) [reply docs/DeveloperGuide.md]: Ok done

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -194,14 +194,46 @@ This section describes some noteworthy details on how certain features are imple
...
+This operation depends on the size of the current person list and will be relatively fast.
+We will then write the information into a .txt file located at `/data/hall.txt` each separated by a new line. 
+
+Given below is a step-by-step usage scenario of how the `export` feature works:
+
+1. The user launches the application and inputs `export email` into the input box.
+
+2. The `LogicManager#execute()` is then called and the input is parsed through `AddressBookParser#parseCommand()`, returning an `ExportCommand`.
+
+3. The `export` command then calls `ExportCommand#execute()`, and calls `Model#getAddressBook()` follwed by `ReadOnlyAddressBook#getPersonList()` to get the current list of persons.
```
Comment (schoolex) [inline docs/DeveloperGuide.md]: There's a typo here, follwed -> followed

**Review by schoolex — COMMENTED** [review]
Almost there, 1 typo to fix

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (yanchenglee98) [reply docs/DeveloperGuide.md]: Good catch, fixed. Thanks.

**Review by schoolex — APPROVED** [review]

**Pull request merged**